### PR TITLE
Bump github actions

### PR DESCRIPTION
**Pull request by Goooler**

### CODE REVIEW

**Review by JLLeitschuh — APPROVED** [review]
Looks good to me. @Tapchicoma feel free to merge if you're good with this change. Thanks @Goooler!

File: .github/workflows/build-and-check.yml [hunk 1]
```diff
@@ -15,13 +15,14 @@ jobs:
...
     runs-on: ${{ matrix.os }}
     steps:
-      - uses: actions/checkout@v1
+      - uses: actions/checkout@v2
       - name: set up JDK 1.8
-        uses: actions/setup-java@v1
+        uses: actions/setup-java@v2
         with:
-          java-version: 1.8
+          java-version: 8
```
Comment (nuhkoca) [inline .github/workflows/build-and-check.yml:22]: Can't `JDK 11` be used instead?

Comment (Goooler) [reply .github/workflows/build-and-check.yml:22]: Sure, we can.

Comment (JLLeitschuh) [reply .github/workflows/build-and-check.yml:22]: I'd prefer if we continued to run with JDK 1.8 as a minimum for now and try adding JDK 11 as a matrix version. I want to make sure we don't break JDK 1.8 support, but also testing for JDK 11 support would be good.

Comment (JLLeitschuh) [reply .github/workflows/build-and-check.yml:22]: Don't feel obligated to add JDK 11 testing support in this PR. If you just want to leave it at version 8, that's fine.

Comment (nuhkoca) [reply .github/workflows/build-and-check.yml:22]: @JLLeitschuh sure thing! We can definitely keep it out of scope of this PR but I'd like to take it up to support `JDK 11`, too. WDYT?

Comment (Tapchicoma) [reply .github/workflows/build-and-check.yml:22]: Personally, I would prefer leaving here JDK 1.8 same as Gradle minimal supported JDK version.

**Pull request merged**